pkg/features/features_test.go

```diff
@@ -57,24 +57,38 @@ func testFlagEnabled(t *testing.T, feature FeatureFlag, test envTest, defaultVal
 		}
 
 		got := feature.Enabled()
-		if buildinfo.ReleaseBuild {
+		if buildinfo.ReleaseBuild && unchangeableFeature {
 			assert.Equal(t, got, defaultValue)
 		} else {
 			assert.Equal(t, got, test.expected)
 		}
 	})
 }
 
-func TestFlags(t *testing.T) {
+func TestFeatureFlags(t *testing.T) {
 	assert.Panics(t, func() {
 		registerFeature("blah", "NOT_ROX_WHATEVER", false)
 	})
 	defaultTrueFeature := registerFeature("default_true", "ROX_DEFAULT_TRUE", true)
 	for _, test := range defaultTrueCases {
-		testFlagEnabled(t, defaultTrueFeature, test, true)
+		testFlagEnabled(t, defaultTrueFeature, test, true, false)
 	}
 	defaultFalseFeature := registerFeature("default_false", "ROX_DEFAULT_FALSE", false)
 	for _, test := range defaultFalseCases {
-		testFlagEnabled(t, defaultFalseFeature, test, false)
+		testFlagEnabled(t, defaultFalseFeature, test, false, false)
+	}
+}
+
+func TestUnchangeableFeatureFlags(t *testing.T) {
+	assert.Panics(t, func() {
+		registerUnchangeableFeature("blah", "NOT_ROX_WHATEVER", false)
+	})
+	defaultTrueFeature := registerUnchangeableFeature("default_true", "ROX_DEFAULT_TRUE", true)
+	for _, test := range defaultTrueCases {
+		testFlagEnabled(t, defaultTrueFeature, test, true, true)
```

Comment:
There's still some duplication, but I'm hoping this is easier to read now. Trying to avoid the last bit of copy-paste will just make it harder to follow